### DISCUSSION TITLE
Is there a better way to "wait" for elements that load dynamically or is this one right?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#17038] [Questions and Help]
I have a series of dropdown and sections that are loaded dynamically depending on the previously selected option. 
That means after selecting the first option I need to wait for it to load the second dropdown. In order to avoid using "waits" I'm doing it like this: 

```cy.get('span').should('contain', 'Please select').contains('Please select')``` 

So at first it will retrieve all spans, without the new one cause it didn't load yet. 
I have to add an assertion so it retries the cy.get() command otherwise it will look only in the first set spans it retrieved. 
When it finds a group of spans, containing "Please select" somewhere, then it returns the specific one that "contains('Please select')"

It does work and I'm avoiding the wait but I'm just wondering if that is the "good practice"

**Comment by pwpatton** [comment]
I'm not certain on the last contains but `cy.get('span').should('contain', 'Please select')` should do it. And I think this is good practice yes.

**Reply by MatiDiez** [reply]
Thanks yeah. The only problem is that if the assertion passes, the function will return all spans instead of that specific one. Hence the second thing, but yeah it does work for now :D. Thank you!
